File: 2000 98 & ME/QuickStart/howto/samples/net/WebRequests/VB/clientpost.vb
Imports System
Imports System.Net
Imports System.IO

Namespace Client
  Public class ClientPOST         
    Public Shared Sub  Main()
      Dim args As String ()

   		args = Environment.GetCommandLineArgs()

    	If (args.Length < 2)
        showusage()
        return
			End If

      getPage(args(1), "sl=foo&s2=bar")
      return
      
    End Sub

	public Shared Sub showusage()
		Console.WriteLine("Attempts to POST to a URL")
        Console.WriteLine("Usage:")
        Console.WriteLine("ClientPOST URL")
        Console.WriteLine("Examples:")
        Console.WriteLine("ClientPOST http://www.nba.com")
	End Sub

	public Shared Sub getPage(url As String, payload as String)

		Dim req As WebRequest		
		Dim RequestStream As Stream
		req = WebRequestFactory.Create(url)
		req.Method = "POST"
		req.ContentType = "application/x-www-form-urlencoded"
		Dim SomeBytes() as Byte
		if payload <> Nothing
			SomeBytes = System.Text.Encoding.ASCII.GetBytes(payload)
			req.ContentLength = SomeBytes.Length
			RequestStream = req.GetRequestStream()
			RequestStream.Write(SomeBytes, 0, SomeBytes.Length)
			RequestStream.Close()
		Else
			req.ContentLength = 0
		End if

		Dim result As WebResponse
		Dim ReceiveStream As Stream
		result = req.GetResponse()		
        ReceiveStream = result.GetResponseStream()                
        Console.WriteLine("Response stream received, Status code: " & result.Status)
        
			Dim read(512) As Byte
        	Dim bytes As Integer
        	bytes = ReceiveStream.Read(read, 0, 512)
        	Console.WriteLine("HTML...")
        
        	Do while (bytes > 0) 
        	
            	Console.Write(System.Text.Encoding.ASCII.GetString(read, 0, bytes))
           		bytes = ReceiveStream.Read(read, 0, 512)
        	Loop

        	Console.WriteLine("")

	End Sub

    End Class
End Namespace

File: 2000 98 & ME/QuickStart/howto/samples/net/WebRequests/clientpost.cs

using System;
using System.Net;
using System.IO;

class ClientPOST {
    public static void Main(string[] args) {

        if (args.Length < 1) {
            showusage();     
            return;
        }   

        getPage(args[0], "s1=food&s2=bart");        

        return;
    }

    public static void showusage() {
        Console.WriteLine("Attempts to POST into to a URL");
        Console.WriteLine("\r\nUsage:");
        Console.WriteLine("\r\nClientPOST URL");
        Console.WriteLine("\r\nExamples:");
        Console.WriteLine("\r\nClientPOST http://www.microsoft.com");
    }
    
    public static void getPage(String url, String payload) {
        WebRequest req = WebRequestFactory.Create(url);
		req.Method = "POST";                            
        req.ContentType = "application/x-www-form-urlencoded";
        byte[] SomeBytes = null;
        if (payload != null) {
        	SomeBytes = System.Text.Encoding.ASCII.GetBytes(payload);
        	req.ContentLength = SomeBytes.Length;
            Stream newStream = req.GetRequestStream();                                
        	newStream.Write(SomeBytes, 0, SomeBytes.Length);
        	newStream.Close();
        }
        else
        {
        	req.ContentLength = 0;
        }
		WebResponse result = req.GetResponse();
        Stream ReceiveStream = result.GetResponseStream();                
        Console.Write("\nResponse stream received, Status code: " + result.Status);

      	Byte[] read = new Byte[512];
      	int bytes = ReceiveStream.Read(read, 0, 512);

      	Console.WriteLine("\r\nHTML...\r\n");
      
      	while (bytes > 0) 
      	{
          	Console.Write(System.Text.Encoding.ASCII.GetString(read, 0, bytes));
         		bytes = ReceiveStream.Read(read, 0, 512);
      	}

      	Console.WriteLine("");
    }    
};
